modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -688,6 +689,28 @@ private SnapshotRestoreContext prepareContext(
                 if (!F.isEmpty(req.groups()) && !req.groups().contains(grpName))
                     continue;
 
+                if (!skipCompressCheck && meta.isGroupWithCompresion(CU.cacheId(grpName))) {
+                    try {
+                        File path = ctx.pdsFolderResolver().resolveFolders().persistentStoreRootPath();
+
+                        ctx.compress().checkPageCompressionSupported(path.toPath(), meta.pageSize());
```

Comment:
This method also checks that FS is hole-punchable, but to restore we can proceed even on FS without this ability since in any way we don't punch the holes during restore. So we can just check compression processor is enabled (method `checkPageCompressionSupported()` without parameters).

Reply:
Yep, but it will fail later on working grid. It is not safe.